### DISCUSSION TITLE
Require collab example for retriever-aware of private REST Apis/SDKs

### DISCUSSION THREAD

**Discussion by dineshkumarsarangapani** [ShishirPatil/gorilla#90] [General]
Hi @ShishirPatil ,

Thank you for this model, it does wonders. I would like to know how we can leverage gorilla llm for private APIs esp REST APIs.
As your paper suggested, we can either fine-tune or use retriever-aware approach.
Please provide some collab example for retriever-aware approach with REST apis, also some example code on how to fine-tune the model with private apis.

**Comment by ShishirPatil** [comment]
Thanks for starting this discussion @dineshkumarsarangapani The high level idea if I understand correctly, is to use Gorilla in retriever mode during inference, to serve APIs that it wasn't trained on.  While all the details for this is in the paper, this will require me to put in some non-trivial time-commitment to make an easily accessible colab similar to the zero-shot setting. 

If there is a fair amount of interest happy to get to it. Also, happy to mentor if someone from the community wants to take a shot!

**Comment by dineshkumarsarangapani** [comment]
I am ready to create an example if you are ok to mentor.

Let me know how we can proceed.

**Comment by ashishwebt** [comment]
We are planning to build Chatgpt plugin. But we have a lot of Private API which is not optimal if you are creating plugin for chatgpt. We would like to use gorilla llm to call our api behind the scenes to bring data in our plugin using semantic kernel or lang chain. We have open API Specs for all our API.
A example of how to would be helpful and appreciated.

**Comment by davideuler** [comment]
I hope we can leverage gorilla llm for private APIs esp REST APIs too. Could anyone help to give an example?

**Reply by nundys** [reply]
+1
